### PR TITLE
FEATURE: Allow severity-based marker setting

### LINKED ISSUE #21: Feature request: add option to disable "linter infos" on minimap
Linters like Pylint produce "linter infos" if your code doesn't follow the style guide, and this can really clog up the minimap.

A way to disable "linter infos" showing would really help.

### PR DESCRIPTION
**Pull request by zypA13510**
This feature will provide more fine-grained control over the marker style by the level of severity, in addition to a new 'off' marker type which disables marker rendering for that severity completely.

Use case: one may want to display errors as a whole red line, warnings as a small gutter, and no info on the minimap.
image

Close #21.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,17 +19,44 @@
...
       "enum": [
+        "off",
         "line",
         "gutter",
         "highlight-under",
         "highlight-over",
         "highlight-outline"
       ],
-      "description": "Marker type for linter highlights"
+      "description": "Marker type for `severity:Error` highlights"
```
Comment (Arcanemagus) [inline package.json]: Change this to just "for error highlights", users aren't going to know what `severity:Error` means 😉.

File: package.json [hunk 2]
```diff
@@ -19,17 +19,44 @@
...
+      "default": "line",
+      "enum": [
+        "off",
+        "line",
+        "gutter",
+        "highlight-under",
+        "highlight-over",
+        "highlight-outline"
+      ],
+      "description": "Marker type for `severity:Warning` highlights"
```
Comment (Arcanemagus) [inline package.json]: "for warning highlights"

File: package.json [hunk 3]
```diff
@@ -19,17 +19,44 @@
...
+      "default": "gutter",
+      "enum": [
+        "off",
+        "line",
+        "gutter",
+        "highlight-under",
+        "highlight-over",
+        "highlight-outline"
+      ],
+      "description": "Marker type for `severity:Info` highlights"
```
Comment (Arcanemagus) [inline package.json]: "for info highlights"

File: lib/minimap-linter-binding.js [hunk 4]
```diff
@@ -57,12 +74,14 @@ export default class MinimapLinterBinding {
...
     const severity = messageSeverity(message);
-    const minimapMarkerDecoration = this.minimap.decorateMarker(marker, {
-      type: this.markerType,
-      class: `.linter-${severity}`,
-      plugin: 'linter',
-    });
-    this.decorations.add(minimapMarkerDecoration);
+    if (this.markerType[severity] !== 'off') {
+      const minimapMarkerDecoration = this.minimap.decorateMarker(marker, {
+        type: this.markerType[severity],
```
Comment (Arcanemagus) [inline lib/minimap-linter-binding.js]: Not all of the `observe` callbacks could have fired when there are messages available, so all the severities may not be set yet. Uou need to provide a fallback type here (`line`).

Comment (zypA13510) [reply lib/minimap-linter-binding.js]: @Arcanemagus I believe this means an issue in the old code? If the `observe`s were not fired, then `type: this.markerType` would evaluate to `undefined` in the old code?
Anyway, I will fix this in the constructor.

**Comment by zypA13510** [comment]
@Arcanemagus I'm sorry, but the link you provided seems irrelevant? Regardless of that, I have added the migration script to the constructor.
And btw, you should be able to push directly to my repo (probably faster than comments 😁).

File: lib/minimap-linter-binding.js (hunk 4 above)
Comment (Arcanemagus) [reply lib/minimap-linter-binding.js]: Hah, yeah, it looks like the existing code had the same issue 😆

**Comment by Arcanemagus** [comment]
FYI, I fixed the CI builds in #39, you'll need to rebase this on `master` to get a passing build.

**Comment by zypA13510** [comment]
@Arcanemagus done :)

**Comment by zypA13510** [comment]
ping @Arcanemagus

**Comment by Arcanemagus** [comment]
Sorry, for some reason I thought there was still something pending when you last commented. This LGTM, thanks!

**Comment by Arcanemagus** [comment]
Published in v2.2.0, thanks!

**Pull request merged**